### ISSUE TITLE
msg.Sender is not available on solana

### ISSUE THREAD

**Issue by vinodsharmak** [hyperledger/solang#1174]
I am trying to compile my contract but getting 

error: 'msg.sender' is not available on Solana. See https://solang.readthedocs.io/en/latest/targets/solana.html#msg-sender-solana
    ┌─ D:\Github\solana-solidity\ERC20.sol:380:16
    │
380 │         return msg.sender;

Please let me know how I can fix this ? or is there any alternative for "msg.Sender"

**Comment by xermicus** [comment]
Hi @vinodsharmak 

The documentation you linked literally answers your question:

```
The way to implement this on Solana is to have an authority account for the contract that must be a signer for the transaction (note that on Solana there can be many signers too). This is a common construct on Solana contracts.
```

**Comment by seanyoung** [comment]
@vinodsharmak can you show us the solidity you are trying to compile, so we can help you re-write it without `msg.sender`?

**Comment by seanyoung** [comment]
@vinodsharmak you are right, we need an ERC20 example that works on Solana. We should add this to our to do list.
